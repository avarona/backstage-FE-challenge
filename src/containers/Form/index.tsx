import React from 'react';
import styles from './styles.module.scss';
import Input from '../../components/Input';
import Button from '../../components/Button';
import Logger, { Props as LoggerProps } from '../../components/Logger';

import { getDifferenceOfSquares } from '../../api';

type State = {
  inputValue: string;
  resultValue: string;
  log: LoggerProps['log'];
};

class Form extends React.Component<{}, State> {
  state = {
    inputValue: '',
    resultValue: '0',
    log: [],
  }

  onChange = (e: React.FormEvent<HTMLInputElement>) => {
    this.setState({ inputValue: e.currentTarget.value })
  }

  onSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    const { inputValue, log } = this.state;
    const payload = Number(inputValue);

    // make request to backend for the calculation
    getDifferenceOfSquares(payload).then((res: any) => {
      this.setState({ resultValue: res.value, log: [...log, res] })
    }).catch((err) => console.error(err));

    // clear the state
    this.setState({ inputValue: '', resultValue: '' });
  }

  render() {
    const { inputValue, resultValue, log } = this.state;
    return (
      <>
        <div className={styles.formContainer}>
          <div className={styles.resultValue}>
            {resultValue}
          </div>
          <form className={styles.formFields} onSubmit={this.onSubmit} autoComplete="off">
              <Input onChange={this.onChange} value={inputValue} />
              <Button text="Submit" />
          </form>
        </div>

        <Logger log={log}/>
      </>
    )
  }
}

export default Form;
