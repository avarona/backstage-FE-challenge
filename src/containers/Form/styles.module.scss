.formContainer {
  margin: auto;
}

.resultValue {
  font-size: 24px;
}

.formFields {
  display: flex;
}
