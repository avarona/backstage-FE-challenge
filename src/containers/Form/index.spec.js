import React from 'react';
import { shallow } from 'enzyme';
import Form from './';

describe('Form', () => {
  const wrapper = shallow(<Form />);

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('is a form', () => {
    expect(wrapper.find('form').exists()).toBe(true);
  });

  it('has fields', () => {
    expect(wrapper.find('Input').exists()).toBe(true);
    expect(wrapper.find('Button').exists()).toBe(true);
  });
});
