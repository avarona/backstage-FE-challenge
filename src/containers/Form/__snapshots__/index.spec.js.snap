// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `"<div class=\\"formContainer\\"><div class=\\"resultValue\\">0</div><form class=\\"formFields\\" autoComplete=\\"off\\"><div class=\\"inputContainer\\"><input type=\\"text\\" class=\\"input\\" pattern=\\"^[1-9][0-9]?$|^100$\\" name=\\"input\\" value=\\"\\" placeholder=\\"Enter a number between 0 and 101\\" required=\\"\\"/></div><button class=\\"buttonContainer\\" type=\\"submit\\">Submit</button></form></div><ul class=\\"loggerContainer\\"></ul>"`;
